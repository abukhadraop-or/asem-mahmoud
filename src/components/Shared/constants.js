
const NAVBAR_ITEMS = [ 'Movies', 'TV Shows', 'People', 'More'];

export const NAVBAR_SUB_ITEMS = {
  More: ['Discussions', 'Leaderboard', 'Support', 'API'],
  Movies: ['Popular', 'Now Playing', 'Upcoming', 'Top Rated'],
  People:['Popular People'],
  'TV Shows': ['Popular', 'Airing Today', 'On TV', 'Top Rated'],
};

export const APP = {
  FOOTER: {
    LIST: [
      {
        items: ['About TMDB', 'Contact Us', 'Support Forums', 'API', 'System Status'],
        name: 'The Basics',
      },
      {
        items: ['Contribution Bible', 'Add New Movie', 'Add New TV Show'],
        name: 'GET INVOLVED',
      },
      {
        items: ['Guidelines', 'Discussions', 'Leaderboard', 'Twitter'],
        name: 'COMMUNITY',
      },
      {
        items: ['Terms of Use', 'API Terms of Use', 'Privacy Policy'],
        name: 'LEGAL',
      },
    ],
    TITLE: 'Join the Community',
  },
  JOIN: 'Join TMDB',
  LANGUAGE: 'en',
  LOAD_MORE: 'Load More',
  LOGIN: 'Login',
  NAVBAR_ITEMS,
  NAVBAR_SUB_ITEMS,
  POPULAR_TITLE: 'Popular Movies',
  SIDEBAR_ITEMS: [...NAVBAR_ITEMS.slice(0, -1), ...NAVBAR_SUB_ITEMS.More ],
  SORT_FILTER: 'Sort',
  SORT_FILTERS: [
    { name: 'Popularity Descending', value: 'popularity.desc'},
    { name: 'Popularity Ascending', value: 'popularity.asc'},
    { name: 'Rating Descending', value: 'vote_average.desc'},
    { name: 'Rating Ascending', value: 'vote_average.asc'},
    { name: 'Release Date Descending', value: 'primary_release_date.desc'},
    { name: 'Release Date Ascending', value: 'primary_release_date.asc'},
    { name: 'Title (Z-A)', value: 'original_title.desc'},
    { name: 'Title (A-Z)', value: 'original_title.asc'},
  ],
  SORT_RESULT: 'Sort Results By',
};
