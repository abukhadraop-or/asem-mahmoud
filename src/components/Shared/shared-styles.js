
import styled, { createGlobalStyle } from 'styled-components';

import SourceSansPro from 'assets/fonts/SourceSansPro-Regular.ttf';
import SourceSansProBold from 'assets/fonts/SourceSansPro-Bold.ttf';
import SourceSansProLight from 'assets/fonts/SourceSansPro-Light.ttf';
import SourceConsensus from 'assets/fonts/Consensus-Bold.otf';

export const VIEW_PORT = {
  mobileView: '62.5rem',
}

export const CSSReset = createGlobalStyle`
  @font-face {
    font-family: 'Source Sans Pro';
    font-style: normal;
    font-weight: 200;
    src: url(${SourceSansPro}) format('truetype');
  }

  @font-face {
    font-family: 'Source Sans Pro Light';
    font-style: normal;
    font-weight: 200;
    src: url(${SourceSansProLight}) format('truetype');
  }

  @font-face {
    font-family: 'Consensus';
    font-style: normal;
    font-weight: 400;
    src: url(${SourceConsensus}) format('opentype');
  }

  @font-face {
    font-family: 'Source Sans Pro Bold';
    font-style: normal;
    src: url(${SourceSansProBold}) format('opentype');
  }

  :root {
    --tmdb-dark-blue: #152538;
    --tmdb-light-blue: #032541;
    --white-color: #ffffff;
    --black-color: #000;
    --primary-font: 'Source Sans Pro', Arial, sans-serif;
    --light-gray: #ffffff99;
    --dark-gray: #D3D3D3;
    --light-black: #00000026;
    --light-shade: #e3e3e3;
    --darkness-gray: #00000099;
    --dark-green: #081c22;
  }

  *,
  *::before,
  *::after {
    box-sizing: border-box;
    font-family: var(--primary-font);
    margin: 0;
    padding: 0;
  }
`;

export const Container = styled.div`
  margin: 0 auto;
  max-width: calc(100% - 1rem);
  padding: 0 0.5rem;

  @media(min-width: ${VIEW_PORT.mobileView}) {
    max-width: 83.5rem;
    padding: 0 2rem;
  }
`;

export const Items = styled.ul`
  align-items: center;
  display: flex;
  gap: 1rem;
  list-style-type: none;
`;

export const Button = styled.button`
  background: transparent;
  border: none;
  cursor: pointer;
`;
