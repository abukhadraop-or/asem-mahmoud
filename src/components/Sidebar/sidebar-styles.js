import styled from 'styled-components';


export const SidebarWrapper = styled.div`
  background-color: var(--tmdb-light-blue);
  height: 100%;
  left: 0;
  position: fixed;
  top: 3.5rem;
  transition: width 0.5s;
  width: ${({ show }) => (show ? 90 : 0)}%;
  z-index: 1;
`;

export const PrimaryItems = styled.ul`
  display: ${({ show }) => show ? 'block':  'none'};
  list-style-type: none;
  padding: 1rem;
`;

export const SecondaryItems =styled.ul`
  list-style-type: none;
  padding: 0.5rem 0;
`;

export const PrimaryItem = styled.li`
  color: var(--white-color);
  cursor: pointer;
  font-size: 1.25rem;
  padding-bottom: 0.652rem;
`;

export const SecondaryItem = styled.li`
  color: var(--light-gray);
  font-size: 1rem;
  padding-bottom: 0.652rem;
`;
