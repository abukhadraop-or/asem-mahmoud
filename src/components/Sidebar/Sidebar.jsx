import React, { useState } from 'react'
import PropTypes from 'prop-types'

import { APP } from 'components/Shared/constants'
import {
  SidebarWrapper,
  PrimaryItems,
  PrimaryItem,
  SecondaryItems,
  SecondaryItem,
} from './sidebar-styles'


/**
 *  Sidebar list for mobile view.
 * @param {Object} props
 * @param {boolean} props.show Flag for display sidebar.
 * @return {JSX.Element}
 */
export default function Sidebar({ show }) {
  const [selected, setSelected] = useState('')

  /**
   * @param {string} item - The selected item .
   * @return {void}
   */
  const handleSelectedSubMenu = (item) => {
    setSelected(item === selected ? '' : item)
  }

  return (
    <SidebarWrapper show={show}>
      <PrimaryItems show={show}>
        {APP.SIDEBAR_ITEMS.slice(0, 3).map((item) => (
          <PrimaryItem key={item} onClick={() => handleSelectedSubMenu(item)} >
            {item}
            {selected === item && (
              <SecondaryItems>
                {APP.NAVBAR_SUB_ITEMS[`${item}`].map((subItem) => (
                  <SecondaryItem key={subItem}>{subItem}</SecondaryItem>
                ))}
              </SecondaryItems>
            )}
          </PrimaryItem>
        ))}
        {APP.SIDEBAR_ITEMS.slice(3).map((item) => (
          <SecondaryItem key={item}>{item}</SecondaryItem>
        ))}
      </PrimaryItems>
    </SidebarWrapper>
  )
}

Sidebar.propTypes = { show : PropTypes.bool.isRequired }

