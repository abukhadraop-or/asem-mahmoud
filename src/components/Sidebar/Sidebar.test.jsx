/* eslint-disable testing-library/no-node-access */
import React from 'react'
import { shallow } from 'enzyme'
import { APP, NAVBAR_SUB_ITEMS } from 'components/Shared/constants'
import Sidebar from './Sidebar'
import '@testing-library/jest-dom'
import { PrimaryItem, PrimaryItems, SecondaryItems, SidebarWrapper } from './sidebar-styles'


describe('<Sidebar />', () => {
  const show = true
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Sidebar show={show} />)
  })

  it('renders correctly', () => {
    expect( wrapper).toMatchSnapshot();
  });

  it('should SidebarWrapper and PrimaryItems have correct props', () => {
    const sidebarWrapper = wrapper.find(SidebarWrapper)
    const primaryItems = wrapper.find(PrimaryItems)

    expect(sidebarWrapper.prop('show')).toBeTruthy()
    expect(primaryItems.prop('show')).toBeTruthy()
  })


  it('should render correct primary items', () => {
    const primaryItem = wrapper.find(PrimaryItem)

    expect(primaryItem).toHaveLength(APP.SIDEBAR_ITEMS.slice(3).length - 1)
  })


  it('should expand secondary items when click on primary', () => {
    const primaryItem = wrapper.find(PrimaryItem).first()
    const itemName = primaryItem.text()

    primaryItem.simulate('click')
    const secondaryItems = wrapper.find(SecondaryItems)

    expect(itemName).toEqual('Movies')
    expect(secondaryItems.children()).toHaveLength(NAVBAR_SUB_ITEMS[itemName].length)
  })
})
