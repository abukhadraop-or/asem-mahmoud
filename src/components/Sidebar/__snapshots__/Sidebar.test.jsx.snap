// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders correctly 1`] = `
<styled.div
  show={true}
>
  <styled.ul
    show={true}
  >
    <styled.li
      key="Movies"
      onClick={[Function]}
    >
      Movies
    </styled.li>
    <styled.li
      key="TV Shows"
      onClick={[Function]}
    >
      TV Shows
    </styled.li>
    <styled.li
      key="People"
      onClick={[Function]}
    >
      People
    </styled.li>
    <styled.li
      key="Discussions"
    >
      Discussions
    </styled.li>
    <styled.li
      key="Leaderboard"
    >
      Leaderboard
    </styled.li>
    <styled.li
      key="Support"
    >
      Support
    </styled.li>
    <styled.li
      key="API"
    >
      API
    </styled.li>
  </styled.ul>
</styled.div>
`;
